### PR TITLE
Update static auth

### PR DESCRIPTION
**Pull request by OpaL2**
## Content:
* Add support for static authentication by using config files.
* Add support for npm start in production server.
* Remov some unneccecary files.
* Update gitignore, now ignores tarballs.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,3 +52,6 @@ nohup.out
 
 # tests
 test/output/*
+
+# package-lock.json
+package-lock.json
```
Comment (Blizzara) [inline .gitignore]: package-lock.json belongs in the repo. The point is that unlike package.json, it sets the specific versions for both our dependencies and their dependencies, or the whole dependency chain. That makes sure that whoever executes "yarn" (~npm install) while having the same package-lock.json file, gets the exact same packages (that are known to work).

=> Please revert this part.

File: app.ts [hunk 2]
```diff
@@ -13,22 +13,23 @@ var compression = require('compression');
...
 var adminApi = require('./backend/build/routes-admin');
 var checkerApi = require('./backend/build/routes-checker');
 
 var log = require('./backend/build/log.js');
 
+var auth = require('./backend/build/auth.js');
+
 var adminAuth = httpAuth.basic({
-    realm: 'Nappikauppa v2 - use your speksi-intra account'
+    realm: 'Nappikauppa v2 ad'
```
Comment (Blizzara) [inline app.ts]: What's the "ad"? Consider removing or using a whole word...

File: app.ts [hunk 3]
```diff
@@ -13,22 +13,23 @@ var compression = require('compression');
...
 
 var config = require('./config/config.js');
-var auth = require('./backend/build/confluenceAuth.js');
 var api = require('./backend/build/routes');
 var adminApi = require('./backend/build/routes-admin');
 var checkerApi = require('./backend/build/routes-checker');
 
 var log = require('./backend/build/log.js');
 
+var auth = require('./backend/build/auth.js');
```
Comment (Blizzara) [inline app.ts]: I believe this could be without the ".js". Feel free to remove also from log-require.

Also, I wonder if requiring from `backend/build` is the best way for these, but that's another question...

File: frontend/src/js-admin/components/Production.tsx [hunk 4]
```diff
@@ -27,7 +27,8 @@ var PERFORMERS = [{value: 'Teekkarispeksi', name: 'Teekkarispeksi'}, {value: 'N
 var DEFAULT_VALUES: IProduction = {
   performer: 'Teekkarispeksi',
   opens: '1900-01-01T00:00',
-  active: false, /*not checked*/
+  active: false,
```
Comment (Blizzara) [inline frontend/src/js-admin/components/Production.tsx]: Since this is not related, prefer to keep PR's about a single chance. Not really that important here, but it's a good practice overall. No need to revert these now though.

File: config/config-sample.js [hunk 5]
```diff
@@ -20,12 +20,24 @@ module.exports = {
     user: '13466',
     password: '6pKF4jkv97zmqBJ3ZL8gUw5DfT2NMQ'
   },
+  auth_method:"static",
```
Comment (Blizzara) [inline config/config-sample.js]: Here I'd prefer:
auth: {
  method: 'static',
  static: { ...},
  confluence: {...}
}

**Comment by OpaL2** [comment]
Fixed gitignore and removed strange string from auth

File: config/config-sample.js (hunk 5 above)
Comment (Blizzara) [reply config/config-sample.js]: Also note the usage of ' instead of ". Doesn't matter functionally but nicer to keep style constant.

File: .gitignore (hunk 1 above)
Comment (Blizzara) [reply .gitignore]: Actually there is no package-lock.json yet in the repo, but we could add it at some point. It was yarn.lock that I added yesterday... (yarn.lock for yarn, package-lock.json for npm)

**Comment by OpaL2** [comment]
Now silent user check works with static auth.

File: config/config-sample.js [hunk 6]
```diff
@@ -20,12 +20,26 @@ module.exports = {
...
+  auth: {
+    method: 'static',
     groups: {
       admin: 'lippukauppa-admin',
       checker: 'lippukauppa-tarkistin'
+    },
+    confluence: {
+      url: 'http://localhost:3010/groups/'
+    },
+    static_auth: { //must have _auth, static is reserved word
```
[the remark this answers is not in the document]
Comment (OpaL2) [reply config/config-sample.js]: Thanks for pointing this out and actually it is allowed by [MDN JS Lexical grammar](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords):

> Reserved words actually only apply to Identifiers (vs. `IdentifierNames`) . As described in [es5.github.com/#A.1](http://es5.github.io/#A.1), these are all `IdentifierNames` which do not exclude `ReservedWords`.
>```js
> a.import
> a['import']
> a = { import: 'test' }
> ```

File: backend/src/staticAuth.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export function authenticate(user: string, password: string, requiredGroup: string, cb: Function) {
+
+  log.info('Trying to authenticate', {user: user});
+
+  var userObject = _.findWhere<IStaticAuthUser, Object>(
+    config.auth.static.users,
+    { name: user, pass: md5(password) }
+    );
```
Comment (Blizzara) [inline backend/src/staticAuth.ts]: Fix indent (-2 spaces), the closing brace should be on the same level as the line where the opening brace is.

File: backend/src/staticAuth.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+  if (userObject &&
+    (requiredGroup === undefined ||
+      _.contains(userObject.groups, requiredGroup)
+    )) {
+     log.info('Authencation successful', {user: user});
```
Comment (Blizzara) [inline backend/src/staticAuth.ts]: These lines also seem to have weird indent, check
